share/shwap/p2p/bitswap/bitswap.go

```diff
@@ -112,13 +118,35 @@ func NewServer(
 		server.WithTargetMessageSize(targetMessageSize),
 		server.MaxOutstandingBytesPerPeer(outstandingBytesPerPeer),
 	)
+	return server.New(ctx, net, bstore, opts...)
+}
 
-	return server.New(
-		ctx,
-		net,
-		bstore,
-		opts...,
+// New constructs Bitswap client and server with parameters optimized for Shwap protocol composition.
+func New(
+	ctx context.Context,
+	net network.BitSwapNetwork,
+	bstore blockstore.Blockstore,
+	opts ...bitswap.Option,
+) *bitswap.Bitswap {
+	opts = append(
+		opts,
+		// client options
+		bitswap.SetSimulateDontHavesOnTimeout(simulateDontHaves),
+		bitswap.ProviderSearchDelay(providerSearchDelay),
+		bitswap.RebroadcastDelay(delay.Fixed(rebroadcastDelay)),
+		// Prevents Has calls to Blockstore for metric that counts duplicates
+		// Unnecessary for our use case, so we can save some disk lookups.
+		bitswap.WithoutDuplicatedBlockStats(),
+
+		// server options
+		bitswap.ProvideEnabled(providesEnabled),
+		bitswap.SetSendDontHaves(sendDontHaves),
+		bitswap.MaxQueuedWantlistEntriesPerPeer(maxServerWantListsPerPeer),
+		bitswap.WithTargetMessageSize(targetMessageSize),
+		bitswap.MaxOutstandingBytesPerPeer(outstandingBytesPerPeer),
 	)
+
```

Comment:
We can avoid duplicating this by defining 
```
type Bitswap {
 Client ...
 Server ...
}

func (b *Bitswap) NotifyNewBlocks() error {
  // calls both
}
```

We don't to define other methods and simply calling `network.Start(client, server)` should be enough to cover other methods.

Reply:
This also makes it easier for to reuse common Server accross two networks.